### ISSUE TITLE
GetAnchorPropertiesAsync behavior with a deleted anchorId

### ISSUE THREAD

**Issue by vincentfretin** [Azure/azure-spatial-anchors-samples#133]
On Android Samsung S8, ASA SDK 2.1.1 on Unity 2019.2.20f1

In the documentation here
https://docs.microsoft.com/en-us/azure/spatial-anchors/how-tos/create-locate-anchors-unity
in the "Update properties" section there is the following snippet:
```
    var anchor = await cloudSession.GetAnchorPropertiesAsync(@"anchorId");
    if (anchor != nullptr)
    {
        anchor.AppProperties[@"last-user-access"] = @"just now";
        await this.cloudSession.UpdateAnchorPropertiesAsync(anchor);
    }
```

What is `nullptr`? Where is it defined? Looking at the SDK code, this may be `IntPtr.Zero`? Can it be null?

In my app, I tried to use cloudSession.GetAnchorPropertiesAsync with an anchorId that I previously deleted with DeleteAnchorAsync in a previous session. I thought it would gave be null, but I got an ArgumentException exception instead that was triggered by the CheckStatus function inside GetAnchorPropertiesAsync.
Is this a normal behavior or is this a bug?

**Comment by nincara** [comment]
I don't know if this is connected but since yesterday I have troubles with my app too. First Authentification Problems like in #132 and now with the AppProperties. The moment I want to get the AppProperties from located anchors, I get following error in Android Logcat. I save all anchors in a list and then loop trough them and spawn them. Sometimes it works for the first anchor, sometimes for two and sometimes nothing happens. Note that this worked some days ago perfectly for weeks. If I comment the lines with AppProperties out, everything works fine and all anchors spawn (I store some data like names and stuff in the AppProperties).  

I work with Unity, Android and a Samsung S9+. 

2020.02.20 13:30:33.519 26800 26826 Error Unity      
at Microsoft.Azure.SpatialAnchors.NativeLibraryHelpers.CheckStatus (System.IntPtr handle, Microsoft.Azure.SpatialAnchors.status value) [0x00101] in <963b7d4f182d4886a56a4819b6b06e22>:0 
2020.02.20 13:30:33.519 26800 26826 Error Unity      
at Microsoft.Azure.SpatialAnchors.IDictionary_String_String.InternalGetItem (System.String key) [0x00014] in <963b7d4f182d4886a56a4819b6b06e22>:0 
2020.02.20 13:30:33.519 26800 26826 Error Unity      
at Microsoft.Azure.SpatialAnchors.BasePrivateDictionary`2[TKey,TValue].get_Item (TKey key) [0x00000] in <963b7d4f182d4886a56a4819b6b06e22>:0 
2020.02.20 13:30:33.519 26800 26826 Error Unity      
at Microsoft.Azure.SpatialAnchors.Unity.MyAppBase.SpawnNewAnchoredObject (UnityEngine.Vector3 worldPos, UnityEngine.Quaternion worldRot, Microsoft.Azure.SpatialAnchors.CloudSpatialAnchor cloudSpatialAnchor) [0x0003f] in <963b7d4f182d4886a56a4819b6b06e22>:0 
2020.02.20 13:30:33.519 26800 26826 Error Unity      
at Microsoft.Azure.SpatialAnchors.Unity.MySpatialApp.<ShowAllAnchors>b__39_0 () [0x00032] in <963b7d4f182d4886a56a4819b6b06e22>:0 
2020.02.20 13:30:33.519 26800 26826 Error Unity      
at Microsoft.Azure.SpatialAnchors.Unity.UnityDispatcher.Update () [0x0003f] in <963b7d4f182d48

**Comment by vincentfretin** [comment]
The problem I describe in this issue is not related to the timeout issue. Currently the service is working, and I reproduced the same issue with an identifier that never existed. It raises an ArgumentException instead of returning null.

**Comment by vincentfretin** [comment]
So currently in my app, instead of having the following code:
```
        public async void RemoveCloudAnchor(string anchorIdentifier)
        {
            try
            {
                var anchor = await CloudManager.Session.GetAnchorPropertiesAsync(anchorIdentifier);
                if (anchor == null)
                {
                    feedbackBox.text = "Anchor already deleted";
                }
                else
                {
                    await CloudManager.DeleteAnchorAsync(anchor);
                }
            }
            catch (Exception ex)
            {
                // Debug.LogError(ex.ToString());  // to show the exception name
                Debug.LogError($"{nameof(ASAManager)} - Error in {nameof(RemoveCloudAnchor)}: {ex.Message} {ex.StackTrace}");
                feedbackBox.text = $"Demo failed, check debugger output for more information";
            }
     }
```

I am using the following code:
```
        public async void RemoveCloudAnchor(string anchorIdentifier)
        {
            try
            {
                var anchor = await CloudManager.Session.GetAnchorPropertiesAsync(anchorIdentifier);
                // GetAnchorPropertiesAsync throws a System.ArgumentException from CheckStatus if the anchor doesn't exist.
                await CloudManager.DeleteAnchorAsync(anchor);
            }
            catch (ArgumentException)
            {
                feedbackBox.text = "Anchor already deleted";
            }
            catch (Exception ex)
            {
                Debug.LogError($"{nameof(ASAManager)} - Error in {nameof(RemoveCloudAnchor)}: {ex.Message} {ex.StackTrace}");
                feedbackBox.text = $"Demo failed, check debugger output for more information";
            }
     }
```

**Comment by vincentfretin** [comment]
This is still an issue with SDK 2.2.0.
I'll add a null check in my code to be sure to handle the case if the issue is fixed in the SDK in a future version.

```
        public async void RemoveCloudAnchor(string anchorIdentifier)
        {
            try
            {
                var anchor = await CloudManager.Session.GetAnchorPropertiesAsync(anchorIdentifier);
                // GetAnchorPropertiesAsync throws a System.ArgumentException from CheckStatus if the anchor doesn't exist.
                if (anchor != null) // just in case they fixes the issue https://github.com/Azure/azure-spatial-anchors-samples/issues/133
                {
                    await CloudManager.DeleteAnchorAsync(anchor);
                }
                else
                {
                   feedbackBox.text = "Anchor already deleted (null)";
                }
            }
            catch (ArgumentException)
            {
                feedbackBox.text = "Anchor already deleted (ArgumentException)";
            }
            catch (Exception ex)
            {
                Debug.LogError($"{nameof(ASAManager)} - Error in {nameof(RemoveCloudAnchor)}: {ex.Message} {ex.StackTrace}");
                feedbackBox.text = $"Demo failed, check debugger output for more information";
            }
     }
```

**Comment by weinanshen** [comment]
Hi @vincentfretin, thanks for letting us know!

The doc is wrong, it should be `if (anchor != null)`. We will fix the documentation [internal issue #25303037].

About ArgumentException, it seems a bug in the SDK. For a non-existed anchor Id, the call should return null. We will investigate it [internal issue #25302947].

Thanks!

**Comment by craigktreasure** [comment]
I've submitted a fix to the docs.
